### PR TITLE
Adds MailChimp Integration

### PR DESCRIPTION
**Pull request by Miserlou**
## Issue Number

https://github.com/AlexsLemonade/refinebio-frontend/issues/324#issuecomment-424762563

## Types of changes

What types of changes does your code introduce?

- New feature (non-breaking change which adds functionality)

## Functional tests

Tried locally, but not tested automatically since we don't want to thrash the list. Please review thoroughly though.

### CODE REVIEW

File: api/data_refinery_api/views.py [hunk 1]
```diff
@@ -250,6 +261,18 @@ def perform_update(self, serializer):
...
+                    client = mailchimp3.MailChimp(mc_user=MAILCHIMP_USER, mc_api=MAILCHIMP_API_KEY)
+                    data = {
+                        "email_address": supplied_email_address,
+                        "status": "subscribed"
+                    }
+                    client.lists.members.create(MAILCHIMP_LIST_ID, data)
+                except mailchimp3.mailchimpclient.MailChimpError as mc_e:
+                    pass # This is likely an user-already-on-list error. It's okay.
+                except Exception as e:
+                    pass # Something outside of our control has gone wrong. It's okay.
```
Comment (kurtwheeler) [inline api/data_refinery_api/views.py]: Could we get a log line here so that if this is happening every time we'll know?

**Comment by Miserlou** [comment]
Sorted

**Comment by arielsvn** [comment]
Will this add all emails to MailChimp?? Because I think we want it to be optional:

image

**Comment by Miserlou** [comment]
Yes, this will add all. Where are you seeing that second checkbox, is that implemented somewhere? I haven't seen that before

**Comment by arielsvn** [comment]
It's not yet implemented, but Deepa updated the [the designs](https://projects.invisionapp.com/d/main#/console/13557531/292620200/preview) today. We could add an additional parameter for this.

**Comment by Miserlou** [comment]
This branch now accepts a `email_ccdl_ok` field, which will be accepted upon job start submission.

**Comment by arielsvn** [comment]
Is it possible to mark these emails somehow when adding them to mailchimp to be able to differentiate them from the ones that subscribed through the landing page? Maybe with a tag?

**Comment by kurtwheeler** [comment]
This has been approved for a while now. Is there something we're waiting on?

**Comment by Miserlou** [comment]
It's been a hot minute but I think this is ready to go again.

File: workers/data_refinery_workers/processors/smasher.py [hunk 2]
```diff
@@ -73,6 +73,7 @@ def _prepare_files(job_context: Dict) -> Dict:
         # Delay failing this pipeline until the failure notify has been sent
         job_context['dataset'].failure_reason = error_message
         job_context['dataset'].success = False
+        job_context['dataset'].is_processing = False
```
Comment (kurtwheeler) [inline workers/data_refinery_workers/processors/smasher.py]: I wouldn't expect to see changes in this file in this PR

**Comment by Miserlou** [comment]
Bad merge, smasher is now from dev

File: common/data_refinery_common/migrations/0014_organism_experiments.py [hunk 3]
```diff
@@ -15,4 +15,4 @@ class Migration(migrations.Migration):
             name='experiments',
             field=models.ManyToManyField(through='data_refinery_common.ExperimentOrganismAssociation', to='data_refinery_common.Experiment'),
         ),
-    ]
+]
```
Comment (kurtwheeler) [inline common/data_refinery_common/migrations/0014_organism_experiments.py]: Is this change on purpose and/or meaningful? I probably wouldn't worry about it but you know python and its whitespace.

Comment (Miserlou) [reply common/data_refinery_common/migrations/0014_organism_experiments.py]: Just my editor trying to be helpful
